src/pages/@hiii.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useRouter } from 'next/router';
+import ValidatorDetails from '@screens/validator_details';
+
+const ProfilePage = () => {
+  const router = useRouter();
+  // console.log('router', router);
+
+  useEffect(() => {
+    const regex = /^\/@/;
+    const regexCheck = regex.test(router.route);
+    // const regexCheck = regex.exec(router.route); //null
+    // const regexCheck = router.route.match(regex);
+    // console.log('regexCheck', regexCheck);
+
+    if (!regexCheck) {
+      router.replace('/');
+    }
+  }, []);
+
+  return (
+    // <ValidatorDetails />
+    <>hiii</>
+  );
+};
+
+export default ProfilePage;
```

Comment:
File name should be `src/pages/[dtag].tsx`. As i've moved all logic inside `screens` folder structure should be respected in most scenarios.

Your starting point should be `src/screens/profile`

Move the logic you have here inside `src/screens/profile/index.ts`.

use `router.query.dtag` instead of `router.route`

Reply:
@ryuash Okay.